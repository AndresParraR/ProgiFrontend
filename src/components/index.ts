export * from './vehicleTotal';
export * from './general';
