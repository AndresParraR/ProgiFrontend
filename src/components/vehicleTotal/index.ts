export { default as TableVehicle } from './TableVehicle.vue';
export { default as FilterVehicle } from './FilterVehicle.vue';
export { default as DialogVehicle } from './DialogVehicle.vue';
