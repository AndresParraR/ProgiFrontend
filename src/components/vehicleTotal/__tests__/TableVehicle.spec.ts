import { mount } from "@vue/test-utils";
import { TableVehicle } from "@/components";
import { beforeEach, describe, expect, test, vi, vitest } from "vitest";
import {
  Dialog,
  Notify,
  QBtn,
  QCard,
  QCardSection,
  QDialog,
  QFile,
  QForm,
  QIcon,
  QSelect,
  QTable,
  QTd,
  QTr,
  Quasar,
} from "quasar";
import { createTestingPinia } from "@pinia/testing";
import { installQuasarPlugin } from "@quasar/quasar-app-extension-testing-unit-vitest";

installQuasarPlugin({
  components: {
    QDialog,
    QCard,
    QCardSection,
    QBtn,
    QTable,
    QIcon,
    QTr,
    QTd,
    QSelect,
    QForm,
    QFile,
  },
  plugins: { Dialog, Notify },
});

describe("TableVehicle Component", () => {
  let wrapper: any = null;

  beforeEach(() => {
    const global = {
      plugins: [
        createTestingPinia({
          createSpy: vitest.fn,
        }),
        Quasar,
      ],
      // mocks: {
      //   $q: {
      //     dialog: vi.fn(),
      //   },
      // },
    };

    wrapper = mount(TableVehicle, {
      global,
    });
  });

  test("Should render correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("Should handle Delete", () => {
    wrapper.vm.handleDeleteVehicle();
    
    expect(wrapper.vm.handleDeleteVehicle).toBeTruthy();
  });
});
