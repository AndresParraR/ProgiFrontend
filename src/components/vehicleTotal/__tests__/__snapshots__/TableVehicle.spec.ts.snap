// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableVehicle Component > Should render correctly 1`] = `
<div
  class="q-card"
>
  <div
    class="q-table__container q-table--cell-separator column no-wrap q-table__card q-table--no-wrap"
  >
    <!---->
    <div
      class="q-table__middle scroll"
    >
      <table
        class="q-table"
      >
        <thead>
          <tr
            class="bg-primary text-white"
          >
            <th
              class="text-left"
            >
              Vehicle Price ($)
            </th>
            <th
              class="text-center"
            >
              Vehicle Type
            </th>
            <th
              class="text-center"
            >
              Basic Fee ($)
            </th>
            <th
              class="text-center"
            >
              Special Fee ($)
            </th>
            <th
              class="text-center"
            >
              Storage Fee ($)
            </th>
            <th
              class="text-center"
            >
              Total ($)
            </th>
            <th
              class="text-center"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      class="q-table__bottom row items-center q-table__bottom--nodata"
    >
      <i
        aria-hidden="true"
        class="q-icon notranslate material-icons q-table__bottom-nodata-icon"
        role="presentation"
      >
        warning
      </i>
      No data available
    </div>
  </div>
</div>
`;
